pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -284,14 +276,13 @@ stdenv.mkDerivation ({
   # dependencies for the build machine.
   #
   # pkgs* arrays defined in stdenv/setup.hs
-  + (optionalString useSeparateSetupDb ''
+  + ''
```

Comment:
@ericson2314 I think this is causing the ARGMAX issue lately - this gives us way more --extra-*-dirs than we used to have. Can we be smarter at avoiding duplicates here? ```buildPkgDb``` will add to arg length much more than ordinary -L stuff.

Reply:
In a later PR, isn't it changed so that `--extra-*` flags are not made for the setup.hs package db?

Reply:
https://github.com/nixos/nixpkgs/blob/master/pkgs/development/haskell-modules/generic-builder.nix#L269-L289

Reply:
OK huh that's really weird because I'd suspect that package db is super small. I think the real issue is having ghc as two *different* types of deps on native causes the setup hook to be run twice, which then without strict deps effectively causes the env hooks to be run twice.